tests/elasticity_test.py

```diff
@@ -0,0 +1,487 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for google3.third_party.py.jax_md.dynamics."""
+
+import numpy as onp
+
+from absl.testing import absltest
+from absl.testing import parameterized
+
+from jax.config import config as jax_config
+from jax import random
+from jax import jit
+from jax import lax
+from jax import grad
+import jax.numpy as np
+
+from jax import test_util as jtu
+
+from jax_md import space
+from jax_md import energy 
+from jax_md import minimize
+from jax_md import quantity
+from jax_md import elasticity
+from jax_md import test_util
+from jax_md.util import *
+
+jax_config.parse_flags_with_absl()
+jax_config.enable_omnistaging()
+FLAGS = jax_config.FLAGS
+
+PARTICLE_COUNT = 64
+STOCHASTIC_SAMPLES = 1
+SPATIAL_DIMENSION = [2, 3]
+
+LOWPRESSURE = [True, False]
+
+if FLAGS.jax_enable_x64:
+  #DTYPE = [f64]
+  DTYPE = [f32, f64]
+else:
+  DTYPE = [f32]
+
+def run_minimization_while(energy_fn, R_init, shift, max_grad_thresh = 1e-12, max_num_steps=1000000, **kwargs):
+  init,apply=minimize.fire_descent(jit(energy_fn), shift, **kwargs)
+  apply = jit(apply)
+
+  @jit
+  def get_maxgrad(state):
+    return jnp.amax(jnp.abs(state.force))
+
+  @jit
+  def cond_fn(val):
+    state, i = val
+    return jnp.logical_and(get_maxgrad(state) > max_grad_thresh, i<max_num_steps)
+
+  @jit
+  def body_fn(val):
+    state, i = val
+    return apply(state), i+1
+
+  state = init(R_init)
+  state, num_iterations = lax.while_loop(cond_fn, body_fn, (state, 0))
+
+  return state.position, get_maxgrad(state), num_iterations
+
+
+class ElasticityTest(jtu.JaxTestCase):
+  # pylint: disable=g-complex-comprehension
+  
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {
+          'testcase_name': '_dim={}_dtype={}_lowpressure={}'.format(dim, dtype.__name__, low_pressure),
+          'spatial_dimension': dim,
+          'dtype': dtype,
+          'low_pressure': low_pressure
+      } for dim in SPATIAL_DIMENSION for dtype in DTYPE for low_pressure in LOWPRESSURE))
+  def test_EMT_from_database_basic(self, spatial_dimension, dtype, low_pressure):
+    if spatial_dimension == 2:
+      N = 64
+    else:
+      N = 128
+
+    if dtype == jnp.float32:
+      max_grad_thresh = 1e-5
+      atol = 1e-4
+      rtol = 1e-3
+    else:
+      max_grad_thresh = 1e-10
+      atol = 1e-8
+      rtol = 1e-5
+
+    for index in range(STOCHASTIC_SAMPLES):
+      cijkl, R, sigma, box = test_util.load_nc_data(spatial_dimension, low_pressure, dtype, index) 
```

Comment:
At the moment this will always load the same element of the database right? Is that fine?

Reply:
Yes. Clearly `STOCHASTIC_SAMPLES` is not the right word. But I'd argue that even if you use an RNG for different initial positions, its still the same set of initial positions every time you run the tests (at least on the same computer, etc.). I changed it to `NUM_SAMPLES`.